src/sql/base/browser/ui/table/plugins/cellSelectionModel.plugin.ts

```diff
@@ -228,29 +239,33 @@ export class CellSelectionModel<T> implements Slick.SelectionModel<T, Array<Slic
 		return newRanges;
 	}
 
-	private handleIndividualCellSelection(e: MouseEvent, args: Slick.OnClickEventArgs<T>) {
-		if (!e.ctrlKey) {
-			return;
-		}
-
-		let ranges: Array<Slick.Range>;
-
-		ranges = this.getSelectedRanges();
-
-		let selectedRange: Slick.Range;
-		if (args.cell === 0) {
-			selectedRange = new Slick.Range(args.row, 1, args.row, args.grid.getColumns().length - 1);
+	private handleCellClick(e: MouseEvent, args: Slick.OnClickEventArgs<T>) {
+		const activeCell = this.grid.getActiveCell();
+		const columns = this.grid.getColumns();
+		const isRowSelectorClicked: boolean = this.options.hasRowSelector && args.cell === 0;
+
+		let newlySelectedRange: Slick.Range;
+		// The selection is a range when there is an active cell and the SHIFT key is pressed.
+		if (activeCell !== undefined && e.shiftKey) {
+			// When the row selector cell is clicked, the new selection is all rows from current active row to target row.
+			// Otherwise, the new selection is the cells in the rectangle between current active cell and target cell.
+			newlySelectedRange = isRowSelectorClicked ? new Slick.Range(activeCell.row, columns.length - 1, args.row, 1)
+				: new Slick.Range(activeCell.row, activeCell.cell, args.row, args.cell);
 		} else {
-			selectedRange = new Slick.Range(args.row, args.cell);
-
+			// If the row selector cell is clicked, the new selection is all the cells in the target row.
+			// Otherwise, the new selection is the target cell
+			newlySelectedRange = isRowSelectorClicked ? new Slick.Range(args.row, 1, args.row, columns.length - 1)
+				: new Slick.Range(args.row, args.cell, args.row, args.cell);
 		}
-		ranges = this.insertIntoSelections(ranges, selectedRange);
 
-		this.grid.setActiveCell(selectedRange.toRow, selectedRange.toCell);
-		this.setSelectedRanges(ranges);
+		// When the CTRL key is pressed, we need to merge the new selection with the existing selections.
+		const rangesToBeMerged: Slick.Range[] = this.isMultiSelection(e) ? this.getSelectedRanges() : [];
+		const result = this.insertIntoSelections(rangesToBeMerged, newlySelectedRange);
+		this.setSelectedRanges(result);
 
-		e.preventDefault();
-		e.stopImmediatePropagation();
+		// Find out the new active cell
```

Comment:
Would be good to explain this logic in the comment too

Reply:
done